eclair-core/src/test/scala/fr/acinq/eclair/payment/ChannelSelectionSpec.scala

```diff
@@ -41,82 +31,82 @@ class ChannelSelectionSpec extends AnyFunSuite {
+//
+//    val channelUpdate = dummyUpdate(ShortChannelId(12345), CltvExpiryDelta(10), 100 msat, 1000 msat, 100, 10000000 msat, true)
+//
+//    // nominal case
+//    assert(relayOrFail(ActorRef.noSender, relayPayload, Some(channelUpdate)) === RelaySuccess(ShortChannelId(12345), CMD_ADD_HTLC(ActorRef.noSender, relayPayload.payload.amountToForward, relayPayload.add.paymentHash, relayPayload.payload.outgoingCltv, relayPayload.nextPacket, origin, commit = true)))
+//    // no channel_update
+//    assert(relayOrFail(ActorRef.noSender, relayPayload, channelUpdate_opt = None) === RelayFailure(CMD_FAIL_HTLC(relayPayload.add.id, Right(UnknownNextPeer), commit = true)))
+//    // channel disabled
+//    val channelUpdate_disabled = channelUpdate.copy(channelFlags = Announcements.makeChannelFlags(isNode1 = true, enable = false))
+//    assert(relayOrFail(ActorRef.noSender, relayPayload, Some(channelUpdate_disabled)) === RelayFailure(CMD_FAIL_HTLC(relayPayload.add.id, Right(ChannelDisabled(channelUpdate_disabled.messageFlags, channelUpdate_disabled.channelFlags, channelUpdate_disabled)), commit = true)))
+//    // amount too low
+//    val relayPayload_toolow = relayPayload.copy(payload = onionPayload.copy(amountToForward = 99 msat))
+//    assert(relayOrFail(ActorRef.noSender, relayPayload_toolow, Some(channelUpdate)) === RelayFailure(CMD_FAIL_HTLC(relayPayload.add.id, Right(AmountBelowMinimum(relayPayload_toolow.payload.amountToForward, channelUpdate)), commit = true)))
+//    // incorrect cltv expiry
+//    val relayPayload_incorrectcltv = relayPayload.copy(payload = onionPayload.copy(outgoingCltv = CltvExpiry(42)))
+//    assert(relayOrFail(ActorRef.noSender, relayPayload_incorrectcltv, Some(channelUpdate)) === RelayFailure(CMD_FAIL_HTLC(relayPayload.add.id, Right(IncorrectCltvExpiry(relayPayload_incorrectcltv.payload.outgoingCltv, channelUpdate)), commit = true)))
+//    // insufficient fee
+//    val relayPayload_insufficientfee = relayPayload.copy(payload = onionPayload.copy(amountToForward = 998910 msat))
+//    assert(relayOrFail(ActorRef.noSender, relayPayload_insufficientfee, Some(channelUpdate)) === RelayFailure(CMD_FAIL_HTLC(relayPayload.add.id, Right(FeeInsufficient(relayPayload_insufficientfee.add.amountMsat, channelUpdate)), commit = true)))
+//    // note that a generous fee is ok!
+//    val relayPayload_highfee = relayPayload.copy(payload = onionPayload.copy(amountToForward = 900000 msat))
+//    val originHighFee = origin.copy(amountOut = relayPayload_highfee.payload.amountToForward)
+//    assert(relayOrFail(ActorRef.noSender, relayPayload_highfee, Some(channelUpdate)) === RelaySuccess(ShortChannelId(12345), CMD_ADD_HTLC(ActorRef.noSender, relayPayload_highfee.payload.amountToForward, relayPayload_highfee.add.paymentHash, relayPayload_highfee.payload.outgoingCltv, relayPayload_highfee.nextPacket, originHighFee, commit = true)))
+//  }
+//
+//  test("channel selection") {
```

Comment:
I think we can remove the previous test (it's already covered in `ChannelRelayerSpec`), but don't we want to keep this one? Or migrate it inside `ChannelRelayerSpec`?

Reply:
What shall we do about this last test (`channel selection`)?
Otherwise the PR looks good to me.

Reply:
Done with 3453ba627 (and fixed a bug). I also simplified `ChannelRelay` by using the appropriate projection of `ChannelUpdates`/`NodeChannels`. I believe we could simplify further by only call the channel selection function once at the beginning to sort the channels in order of preference and try them in order instead of calling it at every round.